src/core/ext/transport/chttp2/transport/writing.c

```diff
@@ -229,6 +229,11 @@ bool grpc_chttp2_begin_write(grpc_exec_ctx *exec_ctx,
       now_writing = true;
       t->ping_state.pings_before_data_required =
           t->ping_policy.max_pings_without_data;
+      if (!t->is_client) {
```

Comment:
Would be nice if you could refactor this to a function like 'reset ping strikes'  (just a nit - you don't have to do it).